### ISSUE TITLE
A lot more querries since 1.4?

### ISSUE THREAD

**Issue by davidhellmann** [verbb/navigation#205]
**Description**
I've updated today and was a bit surprised about my query count in the toolbar. This shows me now 700 DB Querries.
Prior 1.4 they are just 140.

I've multiple navigations. Deepest are 3 Levels. 

Code to include:
```
          {% set navigationMain = craft.navigation.nodes('navigationMain').with(['descendants.descendants']).all() ??? null %}
          {% include '_components/navigation/navigationMain.twig' with {
            props: {
              nodes: navigationMain
            }
          } only %}
```

Macro:
```
{#
Navigation
------------------------------------------------------------

{% import '_macros/_macro-navigation.twig' as macroNavigation  %}

{{ macroNavigation.navigation(props) }}
#}

{% macro navigation(props) %}
  {% set defaults = {
    cn: 'c-navigation',
    nodes: null,
    ulClass: '__list',
    liClass: '__listItem',
    spanClass: '__listHeading',
    linkClass: '__listLink',
    divClass: '__subItemsWrapper',
    customTabIndex: 0
  } %}

  {# Merge Default with Options #}
  {% set props = props is defined ? defaults|merge(props) : defaults %}

  <ul class="{{ props.cn ~ props.ulClass }}  {{ props.cn ~ props.ulClass ~ '--level-1' }}">
    {% set tabIndex = props.customTabIndex %}


    {% nav node in props.nodes %}
      {% set level = node.level %}
      {% set triggerTag = 'a' %}
      {% if node.hasDescendants %}{% set triggerTag = 'button' %}{% endif %}

    <li
      class="{{ props.cn ~ props.liClass }}  {{ props.cn ~ props.liClass ~'--level-' ~ level }}  {% if node.classes | length %}{{ node.classes }}{% endif %}">
      {% if not node.url %}
        {% if node.title != '-' %}
          <span
            class="{{ props.cn ~ props.spanClass }}  {{ props.cn ~ props.spanClass ~ '--level-' ~ level }}" {% for attribute in node.customAttributes %}{{ attribute.attribute }}="{{ attribute.value }}"{% endfor %}
          {% if node.hasDescendants %}aria-haspopup="true" aria-expanded="false"{% endif %}>
          {{- node.title -}}
          </span>
        {% endif %}
      {% else %}
        <{{ triggerTag }} href="{{ node.url }}"
        class="{{ props.cn ~ props.linkClass }}  {{ props.cn ~ props.linkClass ~ '--level-' ~ level }}"
        {% if node.hasDescendants %}aria-haspopup="true" aria-expanded="false"{% endif %}
        {# tabindex="{{ tabIndex }}" #}
        {% if node.newWindow %} target="_blank" rel="noopener"{% endif %}
        {% for attribute in node.customAttributes %}{{ attribute.attribute }}="{{ attribute.value }}"{% endfor %}>
        <span class="{{ props.cn ~ '__title' }}">
        {{- node.title -}}
        </span>
        {% for attribute in node.customAttributes %}
          {% if attribute.attribute == 'name-suffix' %}
            <span class="{{ props.cn ~ '__suffix' }}">{{ attribute.value }}</span>
          {% endif %}
        {% endfor %}
        </{{ triggerTag }}>
        {% set tabIndex = '-1' %}
      {% endif %}

      {# Subitems #}
      {% if node.hasDescendants %}
        <div class="{{ props.cn ~ props.divClass }} {{ props.cn ~ props.divClass ~ '--level-' ~ (level + 1) }}">
      {% endif %}
      {% ifchildren %}
        {% set level = node.level + 1 %}

        <ul class="{{ props.cn ~ '__list' }}  {{ props.cn ~ '__list--level-' ~ level }}">
          {% children %}
        </ul>
      {% endifchildren %}
      {% if node.hasDescendants %}
        </div>
      {% endif %}
      </li>
    {% endnav %}
  </ul>
{% endmacro %}

```
When I remove all Navigations my Querries go down to round about 90
Maybe something changed an I've to updated my implementation :)

**Additional info**
- Plugin version: 1.4
- Craft version: Craft Pro 3.5.13.2
- Multi-site: No
- Propagate nodes: No

**Comment by engram-design** [comment]
That seems odd, nothing significant around nodes was actually updated in 1.4.0. Were you on 1.3.31 or earlier before upgrading?

**Comment by engram-design** [comment]
Found the culprit! Fixed in [1.4.1](https://github.com/verbb/navigation/releases/tag/1.4.1)

**Comment by davidhellmann** [comment]
Ah, thanks @engram-design <3
